File: files/400-499/442/CH-442.R
library(tidyverse)
library(readxl)

path <- "400-499/442/CH-442 Matrix Normalization.xlsx"
input <- read_excel(path, range = "C4:F7", col_names = FALSE) %>% as.matrix()
test <- read_excel(path, range = "J4:M7", col_names = FALSE) %>% as.matrix()

result <- input / (max(diag(input)) - min(diag(input)))

all(result == test)
# True

File: files/400-499/442/CH-442.py
import pandas as pd
import numpy as np

path = "400-499/442/CH-442 Matrix Normalization.xlsx"
input = pd.read_excel(path, usecols="C:F", nrows=4, skiprows=3, header=None).to_numpy()
test = pd.read_excel(path, usecols="J:M", nrows=4, skiprows=3, header=None).to_numpy()

diag = np.diag(input)
value_range = np.diag(input).max() - np.diag(input).min()
result = input / value_range

print((result == test).all())
# True
